akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -349,12 +440,79 @@ class CircuitBreaker(
   private def resetBreaker(): Unit = transition(HalfOpen, Closed)
 
   /**
+   * Invokes all onSuccess callback handlers.
+   *
+   * @param start time in nanoseconds of call invocation
+   */
+  private def notifyCallSuccessListeners(start: Long): Unit = if (!successListeners.isEmpty) {
+    val elapsed = System.nanoTime() - start
+    val iterator = successListeners.iterator()
+    while (iterator.hasNext) {
+      val listener = iterator.next()
+      executor.execute(new Runnable {
+        def run() = listener.accept(elapsed)
+      })
```

Comment:
IMO there should be Runnable per listener. Now, the user decides which ExecutionContext should be used for executing listeners. Wrapping all listeners in single Runnable is not the best idea, because if execution time of one of listeners takes much more longer than the others, this one will block others. This could be a great disadvantages of wrapping them into one Runnable.